front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts

```diff
@@ -91,12 +88,15 @@ const useLazyLoadTrains = ({
             rollingStocks!
           );
 
+          // launch the projection of the trains if needed
+          setTrainIdsToProject?.((prev) => new Set([...prev, ...packageToFetch]));
+
           // as formattedSummaries is a dictionary, we replace the previous values with the new ones
           setTrainScheduleSummariesById((prev) => concatMap(prev, newFormattedSummaries));
         }
       }
 
-      setTrainIdsToFetch([]);
+      setTrainIdsToFetch?.([]);
```

Comment:
Could we make it the caller's responsibility to reset the list of train IDs to fetch if they need to? They can do it with a `useEffect()` on `allTrainsLoaded` I think?

Reply:
Done !